src/main/java/teammates/ui/controller/InstructorFeedbackResultsPageData.java

```diff
@@ -189,26 +199,50 @@ public void initForViewByRecipientQuestionGiver(InstructorAttributes instructor,
         buildSectionPanelsForRecipientQuestionGiver(sortedResponses);
     }
     
+    public void initForViewByRecipientGiverQuestion(InstructorAttributes instructor, 
+                                    String selectedSection, String showStats, 
+                                    String groupByTeam) {
+        viewType = ViewType.RECIPIENT_GIVER_QUESTION;
+        this.instructor = instructor;
+        this.selectedSection = selectedSection;
+        this.showStats = showStats;
+        this.groupByTeam = groupByTeam;
+        
+        if (!bundle.isComplete) {
+            // results page to be loaded by ajax instead 
+            buildSectionPanelsForForAjaxLoading(sections);
+            return;
+        }
+        
+        // Note that if the page needs to load by ajax, then responses will be empty too,
+        // therefore the check whether the bundle needs to come before this
+        if (bundle.responses.isEmpty()) {
+            // no responses, nothing to initialize
+            return;
+        }
+        
+        setShouldCollapsed(bundle.responses.size() > RESPONSE_LIMIT_FOR_COLLAPSING_PANEL);
```

Comment:
Would it be a good idea to put these three segments into a function returning a boolean (to terminate or not)?
Seems to be used multiple times.